share/longtail/climbing/crawler.pl

```diff
@@ -0,0 +1,158 @@
+	my $response = $ua->get($url);
+	if ($response->is_success){
+		$tree->parse($response->decoded_content);
+	}
+	else{
+		die $response->status_line;
+	}
+	
+	my $type = $tree->findvalue('//a[@href="/article/AreaTypes"]');
+	
+	unless($type eq "region"){
+		my $title = $tree->findvalue('//title');
+		$title =~ s/ \| theCrag//;
+		my ($place, $styleinfo) = split(', ', $title);
+		$data->{$place}->{"url"}=$url;
+		print "$place: $url \n";
+		$data->{$place}->{"type"}=$type;
+		
+		my @latmeta = $tree->findnodes('//meta[@property="place:location:latitude"]');
+		my @lonmeta = $tree->findnodes('//meta[@property="place:location:longitude"]');
+		my $lat = $latmeta[0]->attr('content');
+		my $lon = $lonmeta[0]->attr('content');
+		$data->{$place}->{"longitude"} = $lon;
+		$data->{$place}->{"latitude"} = $lat;
+		
+		my @breadcrumbs = $tree->findvalues('//div[@id="breadCrumbs"]/ul/li');
+		my $crumbs = "";
+		foreach my $i(1..$#breadcrumbs){
+			$data->{$place}->{"breadcrumbs"}->[$i-1] = $breadcrumbs[$i];
+			$crumbs = $breadcrumbs[$i] . "," . $crumbs;
+			}
+		$crumbs =~ s/,$//;
+		my $numroutes=$tree->findvalue('//a[@title="Search and filter these routes"]');
+		$numroutes = substr($numroutes,1,(index($numroutes,'routes')-2));
+		$data->{$place}->{"number of routes"}=$numroutes;
+
+		my @styles = $tree->findvalues('//span[@class= "style-breakdown" ]/a');
+		foreach my $i(0..$#styles){
+			my $style = $styles[$i];
+			my $percent = substr($style,1,index($style,"%"));
+			my $type = substr ($style,index($style,"%")+1);
+			$type =~ s/^\s+//g;
+			$data->{$place}->{'styles'}->{$type} = $percent; 
+			}
+
+		my @areas = $tree->findnodes('//div[@class = "area"]' );
+		foreach my $i (0..$#areas){
+			$data->{$place}->{'areas'}->[$i]->{'name'} = substr($areas[$i]->findvalue('div[@class="name"]'),0,-5);
+			$data->{$place}->{'areas'}->[$i]->{'routes'} = $areas[$i]->findvalue('div/div[@class="routes"]');
+			$data->{$place}->{'areas'}->[$i]->{'ticks'} = $areas[$i]->findvalue('div/div[@class="ticks"]');
+			$data->{$place}->{'areas'}->[$i]->{'height'} = $areas[$i]->findvalue('div/div[@class="height"]');
+			my @links= $areas[$i]->findnodes('div/a[@href]');
+			$data->{$place}->{'areas'}->[$i]->{'link'} = $links[1]->attr('href');
+		}
+		
+		my $paragraph = "";
+		my $generalstyle = $tree->findvalue('//h1[@class="inline"]/small');
+		$generalstyle =~ s/^\s+//g;
+		$paragraph .= "$generalstyle<br>Number of Routes: $numroutes<br><br>";
+		my @summaryinfo= $tree->findnodes('//div[@class="node-info summary"]');
+		my @descriptioninfo= $tree->findnodes('//div[@class="node-info description"]');
+		if($#summaryinfo == 0){
+			my @summary= $summaryinfo[0]->findvalues('div/div/p');
+			for my $i (0 .. $#summary){
+				$paragraph .= ($summary[$i] . "<br>");
+			}
+		}
+		if($#descriptioninfo == 0){
+			my @description = $descriptioninfo[0]->findvalues('div/div/p');
+			$paragraph .= ($description[0] . "<br>");
+			if($#description>0) {$paragraph .= "...";}
```

Comment:
Can you make sure that the ellipsis falls on the same line?  
![selection_345](https://cloud.githubusercontent.com/assets/1882892/6107779/b5427e26-b03b-11e4-922e-150ef2fb0740.png)